src/schema/sale/end_sale_mutation.js

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+
+import { GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { SaleType } from "schema/sale/index"
+
+export default mutationWithClientMutationId({
+  name: "EndSale",
+  description:
+    "Mark sale as ended.",
+  inputFields: {
+    sale_id: {
+      type: GraphQLString,
+    },
+  },
+  outputFields: {
+    sale: {
+      type: SaleType,
+      resolve: (
+        { sale_id },
+        _,
+        _request,
+        { rootValue: { saleLoader } }
+      ) => saleLoader(sale_id),
+    },
+  },
+  mutateAndGetPayload: (
+    { sale_id },
+    _request,
+    {
+      rootValue: {
+        accessToken,
+        endSaleLoader,
+      },
+    }
+  ) => {
+    if (!accessToken) {
+      return new Error("You need to be signed in to perform this action")
+    }
+
+    return endSaleLoader(sale_id).then(() => ({ sale_id }))
```

Comment:
Tying in to the above comment on resolving the sale from the result of this API call, rather than re-fetching, you can just do `return endSaleLoader(sale_id)` (assuming you get rid of `headers: true`, otherwise you'd need `endSaleLoader(sale_id).then(({ body }) => body)`.